hedera-node/yahcli/src/test/java/com/hedera/services/yahcli/test/commands/accounts/SetupStakeCommandTest.java

```diff
@@ -0,0 +1,272 @@
+            final var exception = assertThrows(
+                    CommandLine.OverwrittenOptionException.class,
+                    () -> parseArgs(typicalGlobalOptions() + " activate-staking -r 100h -r 200h"));
+            assertThat(exception.getMessage()).contains("should be specified only once");
+        }
+
+        @Test
+        void duplicateRewardAccountBalanceOptionsThrowException() {
+            final var exception = assertThrows(
+                    CommandLine.OverwrittenOptionException.class,
+                    () -> parseArgs(typicalGlobalOptions() + " activate-staking -b 100h -b 200h"));
+            assertThat(exception.getMessage()).contains("should be specified only once");
+        }
+
+        @Test
+        void missingOptionValueThrowsException() {
+            final var exception = assertThrows(
+                    CommandLine.ParameterException.class,
+                    () -> parseArgs(typicalGlobalOptions() + " activate-staking -p"));
+            assertThat(exception.getMessage()).contains("Missing required parameter");
+        }
+    }
+
+    @Nested
+    class ScaledAmountValidation {
+        @ParameterizedTest
+        @MethodSource("validScaledAmounts")
+        void validScaledAmountsAccepted(String amount) {
+            // Test that valid scaled amounts with proper suffixes are accepted
+            final var result = parseArgs(typicalGlobalOptions() + " activate-staking -p " + amount);
+            assertCommandHierarchyOf(result, "yahcli", "activate-staking");
+        }
+
+        @ParameterizedTest
+        @MethodSource("validEdgeCaseAmounts")
+        void validEdgeCaseAmountsAccepted(String amount) {
+            final var result = parseArgs(typicalGlobalOptions() + " activate-staking -p " + amount);
+            assertCommandHierarchyOf(result, "yahcli", "activate-staking");
+        }
+
+        @ParameterizedTest
+        @MethodSource("invalidEdgeCaseAmounts")
+        void invalidEdgeCaseAmountsParsedButWillFailOnExecution(String amount) {
+            // These values are accepted during parsing but would fail during execution
+            final var result = parseArgs(typicalGlobalOptions() + " activate-staking -p " + amount);
+            assertCommandHierarchyOf(result, "yahcli", "activate-staking");
+        }
+
+        @Test
+        void spaceAmountThrowsUnmatchedArgumentException() {
+            final var exception = assertThrows(
+                    CommandLine.UnmatchedArgumentException.class,
+                    () -> parseArgs(typicalGlobalOptions() + " activate-staking -p \" \""));
+            assertThat(exception.getMessage()).contains("Unmatched argument");
+        }
+
+        static Stream<String> validScaledAmounts() {
+            return Stream.of(
+                    "1h", "10kh", "100mh", "1000bh", "1H", "10KH", "100MH", "1000BH", "0", "123456789", "999h", "1bh");
+        }
+
+        static Stream<String> validEdgeCaseAmounts() {
```

Comment:
Not a big deal, but we only want to use `@MethodSource` when there's more than one usage of the values (I made the same mistake 😆). Can do in a future PR